tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/tracee/pass/TraceePassImpl.scala

```diff
@@ -0,0 +1,116 @@
+package at.forsyte.apalache.tla.tracee.pass
+
+import at.forsyte.apalache.infra.passes.DerivedPredicates
+import at.forsyte.apalache.tla.tracee._
+import at.forsyte.apalache.infra.passes.Pass.PassResult
+import at.forsyte.apalache.infra.passes.options.{OptionGroup, SourceOption}
+import at.forsyte.apalache.io.ConfigurationError
+import at.forsyte.apalache.io.json.impl.DefaultTagReader
+import at.forsyte.apalache.io.lir.TlaWriterFactory
+import at.forsyte.apalache.tla.assignments.{AssignmentOperatorIntroduction, ModuleAdapter}
+import at.forsyte.apalache.tla.imp.findBodyOf
+import at.forsyte.apalache.tla.imp.src.SourceStore
+import at.forsyte.apalache.tla.lir.transformations.TransformationTracker
+import at.forsyte.apalache.tla.lir.{ModuleProperty, NullEx, TlaModule, TlaVarDecl}
+import at.forsyte.apalache.tla.pp.NormalizedNames
+import com.google.inject.Inject
+import com.typesafe.scalalogging.LazyLogging
+
+/**
+ * The implementation of a trace evaluation encoding pass.
+ *
+ * @author
+ *   Jure Kukovec
+ */
+class TraceePassImpl @Inject() (
+    derivedPreds: DerivedPredicates.Configurable,
+    options: OptionGroup.HasTracee,
+    tracker: TransformationTracker,
+    writerFactory: TlaWriterFactory,
+    sourceStore: SourceStore)
+    extends TraceePass with LazyLogging {
+
+  override def name: String = "TraceePass"
+
+  // TODO: turn requires into proper error handling?
+  override def execute(module: TlaModule): PassResult = {
+
+    val traceJsonFile = options.tracee.trace
+    val traceSource = SourceOption.FileSource(traceJsonFile)
```

Comment:
This can be removed once `HasTrace` is changed to already load a `SourceOption`, and it should support both a `FileSource` and a `StringSource`.

Reply:
But `TraceeCmd.trace` still has to be a `File`, no?

Reply:
I don't see why!

Reply:
Oh sorry, yes, the CLI input itself is a file.

Reply:
But allowing the config to take an `OptionSource` allows us to pass in strings via JSON config input, either loading it from a config source on disk or via RPC data.

Reply:
So the `SourceOption`s come in at the green level here: https://github.com/informalsystems/apalache/blob/main/docs/src/adr/022adr-unification-of-configs-and-options.md#solution

Reply:
I think I got it, commit incoming soon